File: python/logger/http_logger.py
from typing import Any, Dict
from .logger import Logger

class HttpLogger(Logger):
    def request(self, method: str, url: str, payload: Dict[str, Any] | None = None):
        self.info("HTTP Request", {
            "method": method,
            "url": url,
            "request": payload,
        })

    def response(self, method: str, url: str, status: int, payload: Dict[str, Any] | None = None):
        self.info("HTTP Response", {
            "method": method,
            "url": url,
            "status": status,
            "response": payload,
        })

File: typescript/logger/http-logger.ts
import { Logger, LogLevel } from "./logger";

export interface RequestPayload {
  method: string;
  url: string;
  headers?: Record<string, string>;
  body?: unknown;
}

export interface ResponsePayload {
  statusCode: number;
  headers?: Record<string, string>;
  body?: unknown;
}

const logger = Logger.getInstance(LogLevel.Debug);

export const requestLogger = (params: {
  functionName: string;
  message: RequestPayload;
}) => {
  logger.request({
    functionName: params.functionName,
    message: params.message,
  });
};

export const responseLogger = (params: {
  functionName: string;
  message: ResponsePayload;
}) => {
  logger.response({
    functionName: params.functionName,
    message: params.message,
  });
};
